v2/internal/controllers/crd_dataprotection_backuppolicy_test.go

```diff
@@ -0,0 +1,218 @@
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	// Consts for BackupVault
+	identityType := "SystemAssigned"
+	alertsForAllJobFailures_Status := dataprotection.AzureMonitorAlertSettings_AlertsForAllJobFailures_Enabled
+	StorageSetting_DatastoreType_Value := dataprotection.StorageSetting_DatastoreType_VaultStore
+	StorageSetting_Type_Value := dataprotection.StorageSetting_Type_LocallyRedundant
+
+	// Create a backupvault
+	backupvault := &dataprotection.BackupVault{
+		ObjectMeta: tc.MakeObjectMetaWithName("shayvault1"),
+		Spec: dataprotection.BackupVault_Spec{
+			Location: tc.AzureRegion,
+			Tags:     map[string]string{"cheese": "blue"},
+			Owner:    testcommon.AsOwner(rg),
+			Identity: &dataprotection.DppIdentityDetails{
+				Type: &identityType,
+			},
+			Properties: &dataprotection.BackupVaultSpec{
+				MonitoringSettings: &dataprotection.MonitoringSettings{
+					AzureMonitorAlertSettings: &dataprotection.AzureMonitorAlertSettings{
+						AlertsForAllJobFailures: &alertsForAllJobFailures_Status,
+					},
+				},
+				StorageSettings: []dataprotection.StorageSetting{
+					{
+						DatastoreType: &StorageSetting_DatastoreType_Value,
+						Type:          &StorageSetting_Type_Value,
+					},
+				},
+			},
+		},
+	}
+	tc.CreateResourceAndWait(backupvault)
+
+	// Note:
+	// It is mandatory to create a backupvault before creating a backuppolicy
+
+	// Consts for BackupPolicy
+	backupPolicy_ObjectType := dataprotection.BackupPolicy_ObjectType_BackupPolicy
+
+	// Consts for BackupPolicy:AzureBackupRule
+	AzureBackRule_Name := "BackupHourly"
+	AzureBackupRule_ObjectType := dataprotection.AzureBackupRule_ObjectType_AzureBackupRule
+
+	AzureBackupParams_BackupType_Value := "Incremental"
+	AzureBackupParams_ObjectType_Value := dataprotection.AzureBackupParams_ObjectType_AzureBackupParams
+
+	DataStore_DataStoreType_Value := dataprotection.DataStoreInfoBase_DataStoreType_OperationalStore
+	DataStore_ObjectType_Value := "DataStoreInfoBase"
+
+	Schedule_ObjectType_Value := dataprotection.ScheduleBasedTriggerContext_ObjectType_ScheduleBasedTriggerContext
+	Schedule_Timezone_Value := "UTC"
+
+	TaggingCriteria_isDefault_Value := true
+	TaggingCriteria_TaggingPriority_Value := 99
+	TaggingCriteria_TagInfo_TagName_Value := "Default"
+
+	// Consts for BackupPolicy:AzureRetentionRule
+	AzureRetentionRule_Name := "Default"
+	AzureRetentionRule_ObjectType := dataprotection.AzureRetentionRule_ObjectType_AzureRetentionRule
+	AzureRetentionRule_IsDefault := true
+
+	AzureRetentionRule_Lifecycles_DeleteAfter_Duration := "P9D"
+	AzureRetentionRule_Lifecycles_DeleteAfter_ObjectType := dataprotection.AbsoluteDeleteOption_ObjectType_AbsoluteDeleteOption
+	AzureRetentionRule_Lifecycles_SourceDataStore_DataStoreType := dataprotection.DataStoreInfoBase_DataStoreType_OperationalStore
+	AzureRetentionRule_Lifecycles_SourceDataStore_ObjectType := "DataStoreInfoBase"
+
+	// Create a BackupPolicy
+	backuppolicy := &dataprotection.BackupVaultsBackupPolicy{
```

Comment:
Consider moving this into a subtest : [example](https://github.com/Azure/azure-service-operator/blob/main/v2/internal/controllers/crd_storage_storageaccount_20220901_test.go#L47)

Reply:
Thanks for the suggestion.
Currently, it is prone to not using backup vault creation in this file, as discussed with @Bevan in previous comments. 

Let's see how it goes and then come back to it.